alfresco-dynamic-extensions-repo/build.gradle

```diff
@@ -1,91 +1,107 @@
+        testCompile("org.alfresco:alfresco-repository") { transitive = false }
+        testCompile("org.alfresco:alfresco-data-model") { transitive = false }
+
+        testRuntime "org.springframework:spring-beans"
+        testCompile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
+    }
 
-    amp "biz.aQute.bnd:biz.aQute.bnd:${project.ext.bndVersion}"
+    def ampConfig = {
+        from(configurations.blueprint) {
+            into("config/alfresco/module/$project.parent.name/blueprint-bundles/")
+        }
 
-    blueprint("org.eclipse.gemini:org.eclipse.gemini.blueprint.core:${project.ext.geminiVersion}") {transitive = false}
-    blueprint("org.eclipse.gemini:org.eclipse.gemini.blueprint.io:${project.ext.geminiVersion}") {transitive = false}
-    blueprint(project(path: ":blueprint-override", configuration: 'patch')) {transitive = false}
-    blueprint(project(":blueprint-integration")) {transitive = false}
+        from(configurations.bundles) {
+            into("config/alfresco/module/$project.parent.name/standard-bundles")
+        }
 
-    bundles "org.slf4j:slf4j-api:${project.slf4jVersion}"
-    bundles("org.slf4j:slf4j-log4j12:${project.slf4jVersion}") {transitive = false}
-    bundles(project(":control-panel")) {transitive = false}
-    bundles(project(":event-bus")) {transitive = false}
-    bundles 'org.apache.felix:org.apache.felix.log:1.0.1'
+        from(configurations.amp) {
+            into("lib")
+        }
 
-    testCompile "junit:junit:${project.ext.junitVersion}"
-    testCompile "org.mockito:mockito-core:${project.ext.mockitoVersion}"
+        from("${project.projectDir}/module.properties") {
+            expand(project.properties)
+        }
 
-    testCompile "org.springframework:spring-test"
-    testCompile "org.springframework:spring-web"
-    testCompile("org.alfresco.surf:spring-webscripts")
-    testCompile("org.alfresco:alfresco-repository") { transitive = false }
-    testCompile("org.alfresco:alfresco-data-model") { transitive = false }
+        from("${project.parent.projectDir}/shared/src/main/resources") {
+            into("config")
+        }
 
-    testRuntime "org.springframework:spring-beans"
-    testCompile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
-}
+        from("${project.parent.projectDir}/shared/src/main/config") {
+            into("config/alfresco/module/$project.parent.name")
+        }
 
-def ampConfig = {
-    from(configurations.blueprint) {
-        into("config/alfresco/module/$project.name/blueprint-bundles/")
+        extension "amp"
+        destinationDir = file("$buildDir/amps")
     }
 
-    from(configurations.bundles) {
-        into("config/alfresco/module/$project.name/standard-bundles")
-    }
+    def alfrescoAmp = file("${file("$buildDir/amps")}/${project.name}-${project.version}.amp")
 
-    from(configurations.amp) {
-        into("lib")
+    artifacts {
+        archives sourceJar, javadocJar, alfrescoAmp
```

Comment:
Referring directly to a file in `artifacts {}` is not recommended.
You can depend directly on the `amp` task if you move the `amp` task definition above artifacts